lib/internal/fs/streams.js

```diff
@@ -308,15 +308,25 @@ WriteStream.prototype._write = function(data, encoding, cb) {
     });
   }
 
-  fs.write(this.fd, data, 0, data.length, this.pos, (er, bytes) => {
+  const self = this;
+  const len = data.length;
+  let pos = this.pos;
+  let offset = 0;
+  fs.write(this.fd, data, offset, len, pos, function callback(er, bytes) {
     if (er) {
-      if (this.autoClose) {
-        this.destroy();
+      if (self.autoClose) {
+        self.destroy();
       }
       return cb(er);
     }
-    this.bytesWritten += bytes;
-    cb();
+    self.bytesWritten += bytes;
```

Comment:
nit: I'd put the callback in an arrow function variable so we don't have to use the `self` trick.